### DISCUSSION TITLE
Why is className deprecated?

### DISCUSSION THREAD

**Discussion by AdamJ** [cloudscape-design/components#935] [Q&A] [answered]
I'm curious as to why `className` has been deprecated - looking at almost every component where `className` is listed, it calls out that it has been deprecated and that custom CSS is not supported. Is this due to incompatibility with the component structure or more due to a maintenance aspect?

I have used many frameworks where, even though there are suggested practices/principles in place and deviation from patterns is not advised, `className` is still supported in order to provide flexibility to the community. It would be great to have this available as part of Cloudscape in order to provide that flexibility to the community.

**Comment by aquark** [comment]
Based on how we use className, if it were removed it wouldn't stop us from applying necessary styles, but it would make the code more complicated and fragile as we would have to add extra divs to apply these classes.

**Comment by timogasda** [comment, marked as answer]
Custom classes are marked as deprecated for Cloudscape components to discourage CSS overrides. The internal HTML and CSS structure of a Cloudscape component can change with a Cloudscape update. If you have CSS overrides that rely on these internals, Cloudscape updates might inadvertently break your UI without you noticing.
If you want to use custom CSS styles, it's expected that you apply them to your own `div`s (or other HTML elements under your control).

If you're using class names to identify components during tests, we recommend using data test ids.
